src/VCR/Util/CurlHelper.php

```diff
@@ -175,24 +175,39 @@ public static function setCurlOptionOnRequest(Request $request, $option, $value,
             case CURLOPT_HEADER:
             case CURLOPT_WRITEFUNCTION:
             case CURLOPT_HEADERFUNCTION:
+            case CURLOPT_UPLOAD:
                 // Ignore header, file and writer functions.
                 // These options are stored and will be handled later in handleOutput().
                 break;
-            case CURLOPT_READFUNCTION:
```

Comment:
I don't think we can move it. As you see in Guzzle 4 that I just linked to, `CURLOPT_READFUNCTION` is set before `CURLOPT_POSTFIELDS`. Plus this also avoids the problem of needing to set `CURLOPT_INFILESIZE` before `CURLOPT_READFUNCTION`.